src/trs/mod.rs

```diff
@@ -1,23 +1,95 @@
 //! (representation) Polymorphically-typed term rewriting system.
+//!
+//! An evaluatable first-order [Term Rewriting System][0] (TRS) with a [Hindley-Milner type
+//! system][1].
+//!
+//! [0]: https://wikipedia.org/wiki/Hindley–Milner_type_system
+//!      "Wikipedia - Hindley-Milner Type System"
+//! [1]: https://en.wikipedia.org/wiki/Rewriting#Term_rewriting_systems
+//!      "Wikipedia - Term Rewriting Systems"
 
+mod lexicon;
+mod rewrite;
 mod trace;
-mod trs;
-pub use self::trs::{SampleError, TypeError, TRS, TRSParams, TRSSpace};
+pub use self::lexicon::{GeneticParams, Lexicon};
+pub use self::rewrite::RewriteSystem;
```

Comment:
I'm not totally sold on the `rewrite` and `RewriteSystem` nomenclature, though I know that's how I've referred to it in our offline meetings. I also know that module inception is considered poor practice, so perhaps something like `programinduction::rewrite::trs::TRS`?